stroeer/core/v1/article.proto

```diff
@@ -8,133 +8,318 @@ option go_package = "github.com/stroeer/tapir/go/stroeer/core/v1;core";
+
+  // Generic map containing general content and configuration information of
+  // the asset (optional).
+  //
+  // Depending on the type of the asset, this map will contain different
+  // entry sets, e.g. 'source', 'width' and 'height' for images.
+  //
+  // Clients must be resilient to unknown or missing entry sets.
   map<string, string> fields = 3;
+
+  // Technical metadata of the asset (optional).
   Metadata metadata = 4;
 }
 
+// Hierarchical section information of the article till root.
+//
+// For example:  /politik/deutschland/ -> /politik/ -> /
 message Section {
+  // Section path of the article, e.g. /politik/deutschland/ (required).
   string path = 1;
+
+  // Label of the section path, e.g. "Politik"  (TODO: required or optional?).
   string label = 2;
+
+  // Parent section (optional).
   Section parent = 3;
 }
 
+// References to get access to the article on different technical platforms like
+// canonical web or AMP.
 message References {
-  // article's path, e.g. /section/id_$ID/title.html
+  // Path of the article's path, e.g. /section/id_$ID/title.html (required).
   string path = 1;
-  // article's canonical URL, may differ if external, e.g. https://www.giga.de/external.html
+
+  // Canonical URL of the article, may differ if external, e.g. https://www.giga.de/external.html
+  // (required).
   string canonical = 2;
-  // article's AMP URL, no pattern specified yet.
+
+  // AMP URL of the article (optional).
   string amp = 3;
 }
 
+// Body of the article to render on detail pages.
+//
+// see https://github.com/orgs/stroeer/teams/api/discussions/1
 message Body {
+  // Recursive Body nodes.
   repeated BodyNode children = 1;
 }
 
+// Recursive body structure representing all types of possible
+// body nodes inside an article.
+//
+// Clients must be resilient to unknown or missing nodes.
 message BodyNode {
+  // Type of the node, e.g. 'a', 'p', 'ul', 'li', 'text' (required).
   string type = 1;
+
+  // Text of the node (optional).
   string text = 2;
+
+  // Additional information for the node depending on it's type,
+  // e.g. 'href' for 'a' nodes (optional).
   map<string, string> fields = 3;
+
+  // Children of the node, e.g. the 'text' of a 'p' (optional).
   repeated BodyNode children = 4;
-  repeated Element elements = 5; // inline-video, content|rich-links, gallery
+
+  // Elements of the node, e.g. inline videos or galleries (optional).
+  repeated Element elements = 5;
 }
 
+// Main content type of the article.
 enum ContentType {
+  // Not specified.
   CONTENT_TYPE_UNSPECIFIED = 0;
+
+  // A text article.
   CONTENT_TYPE_ARTICLE = 1;
+
+  // TODO: do we have/need images as main types?
   CONTENT_TYPE_IMAGE = 2;
+
+  // A video article.
   CONTENT_TYPE_VIDEO = 3;
+
+  // A gallery article.
   CONTENT_TYPE_GALLERY = 4;
+
+  // An oembed article.
   CONTENT_TYPE_OEMBED = 5;
+
+  // TODO: do we have/need authors as main types?
   CONTENT_TYPE_AUTHOR = 6;
+
+  // TODO: do we have/need agencies as main types?
```

Comment:
this is more due to technical reasons, when we fetch the author/agency data record, this is used to parse and map the type within the adapter. This type is never used for actual content:

```javascript
{
    "systemdata": {
        "contentType": "agency",
        "documentType": "data-record"
    }
}
```

Reply:
But do we need those in the model then @thisismana ?

Reply:
flag as `internal`?